### PR TITLE
Unit Tests for the `get_agent_group` function

### PR DESCRIPTION
**Pull request by LucioDonda**
|Related issue|
|---|
|#11754|

## Description

This PR adds all necessary UT for the get_agent_group function in  `src/shared/agent_op.c`

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux

- Memory tests for Linux
  - [x] Valgrind (memcheck and descriptor leaks check)

- Decoder/Rule tests
  - [x] Added unit tests (for new features)

### CODE REVIEW

File: src/unit_tests/shared/test_agent_op.c [hunk 1]
```diff
@@ -152,6 +153,65 @@ static void test_create_sendsync_payload(void **state) {
     cJSON_Delete(payload);
 }
 
+static void test_get_agent_group(void **state) {
```
Comment (palaciosjeremias) [inline src/unit_tests/shared/test_agent_op.c:156]: We should divide this test in three different ones. i.e.:
**test_get_agent_group_success()**
,
**test_get_agent_group_missing_agent_info()**
and
**test_get_agent_group_missing_group()**
In this way, each different scenario will have its own test and will be easier to sustain.

File: src/unit_tests/shared/test_agent_op.c [hunk 2]
```diff
@@ -152,6 +153,65 @@ static void test_create_sendsync_payload(void **state) {
...
+    char* group;
```
Comment (palaciosjeremias) [inline src/unit_tests/shared/test_agent_op.c:157]: Always initialize pointers
**char* group = NULL;**

File: src/unit_tests/shared/test_agent_op.c [hunk 3]
```diff
@@ -152,6 +153,65 @@ static void test_create_sendsync_payload(void **state) {
...
+    int agent_id = 1;
+    int wdb_sock = -1;
+    size_t size = OS_SIZE_65536 + 1;
+    int res;
+
+    os_calloc(size, sizeof(char), group);
+    snprintf(group, size, "%s","default");
```
Comment (palaciosjeremias) [inline src/unit_tests/shared/test_agent_op.c:164]: In this test **group** is being used as an output buffer (as it is normally used in the methods that call a **test_get_agent_group**) and also as an input buffer to create the json objects. I think we should only use it as an output buffer, and create the json objects with a raw string in the creation.

File: src/unit_tests/shared/test_agent_op.c [hunk 4]
```diff
@@ -152,6 +153,65 @@ static void test_create_sendsync_payload(void **state) {
...
+    /* Correct single group  */
+    expect_value(__wrap_wdb_get_agent_info, id, agent_id);
+    will_return(__wrap_wdb_get_agent_info, agent_info_correct);
+    res = get_agent_group(agent_id, group, size, NULL);
+    assert_string_equal("default", group);
+    assert_int_equal(res, OS_SUCCESS);
+
+    snprintf(group, size, "%s","default,Group2");
+
+    /* return OK, Wrong groups string  */
```
Comment (palaciosjeremias) [inline src/unit_tests/shared/test_agent_op.c:205]: I´m not sure if this test is required. We are verifying if that output is being over-written?

**Review by palaciosjeremias — CHANGES_REQUESTED** [review]

File: src/unit_tests/shared/test_agent_op.c (hunk 4 above)
Comment (LucioDonda) [reply src/unit_tests/shared/test_agent_op.c:205]: Indeed, I was just checking that the correct json response was being written but It doesn't add anything to the other tests. I'll remove it.

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Please, add a screenshot of the results of the tests with Valgrind.

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**